static/app.js

```diff
@@ -42,6 +41,46 @@ $(function() {
         console.log('Geolocation not enabled in this browser')
     }
 
+    let timeoutId;
+
+    $('#term').on('keyup', ()=> {
+        if ($('#search_type').val() == 'by keyword') {
+            clearTimeout(timeoutId);
+            timeoutId = setTimeout(runAutocomplete, 500)
+        } 
+    });
+
+    async function runAutocomplete(term) {
+        $('#auto').remove()
```

Comment:
Place this inside the try/catch to avoid unhandled promise rejection

Reply:
done